woodwork/indexers.py

```diff
@@ -45,3 +45,52 @@ def __getitem__(self, key):
         else:
             # singular value
             return selection
+
+
+class _iLocIndexerAccessor:
+    def __init__(self, data):
+        self.data = data
+        if dd and isinstance(self.data, dd.DataFrame):
+            raise TypeError("iloc is not supported for Dask DataFrames")
+        elif dd and isinstance(data, dd.Series):
+            raise TypeError("iloc is not supported for Dask Series")
+
+    def __getitem__(self, key):
+        selection = self.data.iloc[key]
+        return _process_selection(selection, self.data)
+
+
+class _locIndexerAccessor:
+    def __init__(self, data):
+        self.data = data
+
+    def __getitem__(self, key):
+        selection = self.data.loc[key]
+        return _process_selection(selection, self.data)
+
+
+def _process_selection(selection, original_data):
+    if isinstance(selection, pd.Series) or (ks and isinstance(selection, ks.Series)):
+        # col_name = selection.name
+        # if isinstance(self.ww_data, ww.DataTable) and set(selection.index.values) == set(self.ww_data.columns):
+        #     # return selection as series if series of one row.
+        #     return selection
+        # if isinstance(self.ww_data, ww.DataTable):
+        #     logical_type = self.ww_data.logical_types.get(col_name, None)
+        #     semantic_tags = self.ww_data.semantic_tags.get(col_name, None)
+        # else:
+        logical_type = original_data.ww.logical_type or None
+        semantic_tags = original_data.ww.semantic_tags or None
+        # if semantic_tags is not None:
+        #     semantic_tags = semantic_tags - {'index'} - {'time_index'}
+        selection.ww.init(logical_type=logical_type,
+                          semantic_tags=semantic_tags,
+                          use_standard_tags=original_data.ww.use_standard_tags,
+                          description=original_data.ww.description,
+                          metadata=original_data.ww.metadata)
```

Comment:
Glad we're adding metadata and description in! I think we hadn't been with DataColumn's but do with DataTable, so it'll be good to be consistent.